### PR TITLE
182 add token allow list

### PR DESCRIPTION
**Pull request by RusseII**
https://canary.discord.com/channels/903094151002857492/930939338425008208/963448414572539984 our discussion about this was here. 

Using access control for this seems atypical, but I think it simplifies the code that we need so it makes sense.

### CODE REVIEW

File: contracts/BondFactory.sol [hunk 1]
```diff
@@ -99,6 +113,10 @@ contract BondFactory is IBondFactory, AccessControl {
         ) {
             revert DecimalsOver18();
         }
+        if (isTokenAllowListEnabled) {
```
Comment (RusseII) [inline contracts/BondFactory.sol:116]: I originally had this as a modifier - but was getting stack to deep error :(

**Review by Namaskar-1F64F — CHANGES_REQUESTED** [review]
nice tests. looks good. I couldn't think of a better way to have the two variables two events two modifiers, etc. maybe if we had more permissions we could do something more involved, but this seems 👍 

for comments / natspec things, I think these are important to convey to any user that's looking at the bond factory so I pretend the user is looking at etherscan and wondering what the lists are actually doing without having to look into the code or hunt around for what exactly the allow list means. also there might be spacing issues in my suggestions you should double check. ✅✅

**Review by Namaskar-1F64F — APPROVED** [review]

File: test/BondFactory.spec.ts [hunk 2]
```diff
@@ -178,8 +203,7 @@ describe("BondFactory", async () => {
     it(
       "should mint a very large number of bonds and handle overflow correctly"
     );
-    it("should handle a robust amount of imputs for the bond creation");
-    it("should work with any amount of decimals >= 18");
+    it("should handle a robust amount of inputs for the bond creation");
```
[the remark this answers is not in the document]
Comment (RusseII) [reply test/BondFactory.spec.ts:206]: not sure. I just fixed a spelling mistake here

**Comment by RusseII** [comment]
@Namaskar-1F64F great suggestion of framing things from the users perspective. All comments addressed and fixed!

**Pull request merged**